### PR TITLE
Added option to disable the input

### PR DESCRIPTION
**Pull request by kingllama**
Issue #194

### CODE REVIEW

File: lib/ReactTags.js [hunk 1]
```diff
@@ -399,16 +402,19 @@ ReactTags.propTypes = {
   shouldRenderSuggestions: PropTypes.func,
   removeComponent: PropTypes.func,
   autocomplete: PropTypes.oneOfType([PropTypes.bool, PropTypes.number]),
+  disabled: PropTypes.bool,
```
Comment (ad1992) [inline lib/ReactTags.js]: can you `disabled` false in defaultProps ?

**Comment by kingllama** [comment]
> Since the input is disabled then should we show the input field ?
Wha do you think ?

This is doable, but I think the better solution is to just leave the field disabled. From there, users can hide it on their own with some CSS, such as `ReactTags__tagInputField:disabled { display: none }`

The use case here would be when the user wants to be able to drag and drop the tags or remove the tags but prevent any input. Furthermore, as of now you can effectively block the UI for drag and drop (by not passing in a `handleDrag` function) and delete (again by not passing in a `handleDelete`), but there is no easy way to block the input UI without disabling the entire component through `readOnly`.

My specific scenario in this case was that we were doing conditional validation, and for consistency, wanted the entire field to "grey out". That way it would be obvious to the end user that they are not able to edit the feild at this time, but it is something that can be editable and not just a list of boxes.

**Comment by ad1992** [comment]
@kingllama can you also change the placeholder to `The input field is disabled` by default in case `disabled` is true and `placeholder` props is not passed ? And also add a test for the same.

**Comment by ad1992** [comment]
@kingllama any updates?

File: lib/ReactTags.js [hunk 2]
```diff
@@ -30,6 +35,7 @@ class ReactTags extends Component {
     this.resetAndFocusInput = this.resetAndFocusInput.bind(this);
     this.handleSuggestionHover = this.handleSuggestionHover.bind(this);
     this.handleSuggestionClick = this.handleSuggestionClick.bind(this);
+    this.getPlaceholderText = this.getPlaceholderText.bind(this);
```
Comment (ad1992) [inline lib/ReactTags.js]: no need to bind

File: lib/ReactTags.js [hunk 3]
```diff
@@ -63,6 +69,18 @@ class ReactTags extends Component {
     });
   }
 
+  getPlaceholderText() {
+    if (this.props.placeholder !== undefined) {
```
Comment (ad1992) [inline lib/ReactTags.js]:
```js
const { placeholder, disabled } = this.props;
    return placeholder
      ? placeholder
      : disabled ? DISABLED_PLACEHOLDER : DEFAULT_PLACEHOLDER;
```

File: test/reactTags.test.js [hunk 4]
```diff
@@ -245,6 +245,29 @@ describe('Test ReactTags', () => {
     $input.simulate('keyDown', { keyCode: DOWN_ARROW_KEY_CODE });
   });
 
+  test('disables the input with disabled prop', () => {
```
Comment (ad1992) [inline test/reactTags.test.js]: `should disable the input when disabled prop is true`

Comment (ad1992) [reply test/reactTags.test.js]: can you actually put all the tests with the `disabled` prop inside a `describe` block ?
`describe('tests disabled prop')`

File: test/reactTags.test.js [hunk 5]
```diff
@@ -245,6 +245,29 @@ describe('Test ReactTags', () => {
...
+        disabled: true,
+      })
+    );
+
+    const $input = $el.find('.ReactTags__tagInputField');
+
+    expect($input.prop('disabled')).to.be.true;
+  });
+
+  test('default placeholder text is different when disabled', () => {
```
Comment (ad1992) [inline test/reactTags.test.js]: `should show DISABLED_PLACEHOLDER when input field is disabled`

File: test/reactTags.test.js [hunk 6]
```diff
@@ -245,6 +245,29 @@ describe('Test ReactTags', () => {
...
+    const $el = mount(
+      mockItem({
+        disabled: true,
+      })
+    );
+
+    const $input = $el.find('.ReactTags__tagInputField');
+    expect($input.prop('placeholder')).to.equal('This input field is disabled');
```
Comment (ad1992) [inline test/reactTags.test.js]: import `constants.js` here and use it directly

File: test/reactTags.test.js [hunk 7]
```diff
@@ -245,6 +245,29 @@ describe('Test ReactTags', () => {
...
+  });
+
```
Comment (ad1992) [inline test/reactTags.test.js]: Can you also add a test inside the describe block for `disabled`
* when disabled prop is false, default placeholder is shown
* when placeholder prop is passed, placeholder prop is shown

File: README.md [hunk 8]
```diff
@@ -350,6 +351,10 @@ Useful for enhancing data entry workflows for your users by ensuring the first m
 
 This option has no effect if there are no [`suggestions`](#suggestionsOption).
 
+<a name="disabled"></a>
+##### disabled (optional)
+Disables the input box, preventing text from being entered.
```
Comment (ad1992) [inline README.md]: `Disables the input field, preventing text from being entered.`

**Pull request closed without merging**